### PR TITLE
feature: Slower turns near obstructions

### PR DESCRIPTION
**Pull request by grieneis**

### CODE REVIEW

File: base_local_planner/include/base_local_planner/speed_limiter.h [hunk 1]
```diff
@@ -170,4 +152,4 @@ class SpeedCostFunction: public base_local_planner::TrajectoryCostFunction {
 };
 
 } /* namespace base_local_planner */
-#endif /* SPEED_COST_FUNCTION_H_ */
+#endif /* SPEED_LIMITER_FUNCTION_H_ */
```
Comment (mrousseau65) [inline base_local_planner/include/base_local_planner/speed_limiter.h]: This should match the #define at the beginning of the file (i.e. SPEED_LIMITER_H_)

File: base_local_planner/src/speed_limiter.cpp [hunk 2]
```diff
@@ -198,23 +212,18 @@ void SpeedCostFunction::calculateFootprintBounds(std::vector<geometry_msgs::Poin
...
+    circumscribed_radius_ = std::numeric_limits<double>::max();
+
     for (auto pt : footprint)
     {
       if (pt.x < footprint_min_x_) {footprint_min_x_ = pt.x;}
       if (pt.x > footprint_max_x_) {footprint_max_x_ = pt.x;}
       if (pt.y < footprint_min_y_) {footprint_min_y_ = pt.y;}
       if (pt.y > footprint_max_y_) {footprint_max_y_ = pt.y;}
+      double dist = std::sqrt(pt.x * pt.x + pt.y * pt.y);
+      if (dist < circumscribed_radius_) {circumscribed_radius_ = dist;}
```
Comment (mrousseau65) [inline base_local_planner/src/speed_limiter.cpp]: Isn't this finding the inscribed radius?

**Review by mrousseau65 — CHANGES_REQUESTED** [review]
Need clarification on the circumscribed radius (either calculation is wrong or variable name is wrong)

File: base_local_planner/src/speed_limiter.cpp (hunk 2 above)
Comment (grieneis) [reply base_local_planner/src/speed_limiter.cpp]: Good catch.  I'm not even quite finding the inscribed radius, just some number in between.  Updating it in the next commit.

File: base_local_planner/include/base_local_planner/speed_limiter.h (hunk 1 above)
Comment (grieneis) [reply base_local_planner/include/base_local_planner/speed_limiter.h]: Fixed.

**Review by mrousseau65 — APPROVED** [review]

**Pull request merged**